COM-10: Bare Bones Shopify Next App

akarshagrawal opened the pull request:
## Description
COM-10, COM-11
Created a barebones app with Next.Js and Typescript, Added the connection with Shopify using .env / shopify-cli.yml configuration files.

## Database schema changes
- Not applicable

## Tests
### Automated test cases added
- test checking the Welcome message in heading

### Manual test cases run
- No manual test

akarshagrawal committed:
bare bones app created

Next js app created with typescript and TDD tests to check using Jest.

akarshagrawal committed:
Added Shopify Conenction

- Added Server Index file to add Shopify Connection
- Added .env.example file to create a similar .env file to connect to the shopify store.

bbbneo333 commented on server/index.ts:
```diff
@@ -0,0 +1,126 @@
+
+dotenv.config()
+
+const {
+  PORT,
+  NODE_ENV,
+  SHOPIFY_API_KEY,
+  SHOPIFY_API_SECRET,
+  SCOPES,
+  HOST
+} = process.env as any
```
Instead of directly using `process.env`, can we try to look into a dedicated configuration package? Would recommend this - https://www.npmjs.com/package/config

The idea behind using this package is:
- You can easily create new environments (staging / production / testing / etc)
- As you have a centralised place for obtaining the config, it becomes easier to mock it during testing.

bbbneo333 commented on server/index.ts:
```diff
@@ -0,0 +1,126 @@
+  router.post("/webhooks", async (ctx) => {
+    try {
+      await Shopify.Webhooks.Registry.process(ctx.req, ctx.res);
+      console.log(`Webhook processed, returned status code 200`);
```
Instead of using `console.log`, can we use a logging library instead?
Would recommend - https://www.npmjs.com/package/winston, as interface is quite easy and support multiple transports.

akarshagrawal replied:
noted, would be using it for logging purposes in the app itself.

bbbneo333 commented:
- Please update the PR title to - `COM-10: Bare Bones Shopify Next App`
- add `.DS_STORE` to `.gitignore`
- Please look into how we can configure eslint. Start point would be - https://www.npmjs.com/package/eslint-config-airbnb
- You'd have to remove dotenv related files and configuration once `config` module is integrated.
- Updated README as well to reflect what new configurations and scripts that have ben added in order to aid a new developer. Please have a look here - https://github.com/jalantechnologies/boilerplate-node-ts#configuration. This is an example of how README can look like.

akarshagrawal committed:
DS Store file removed

akarshagrawal committed:
eslint with airbnb implemented

- Eslint added to the project
- AirBnb rules followed
- Some additional Next/React rules added for linting

akarshagrawal committed:
added config package for environment creation

- added config package for environment creation
- in process of removing dotenv
